content/spire/try/oidc_federation_aws.md

```diff
@@ -0,0 +1,399 @@
+    23456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JW
+    T-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN012
+    3456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT
+    -TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN012
+bundle(spiffe://spire-oidc.example.org):
+    {
+    "keys": [
+        {
+            "use": "jwt-svid",
+            "kty": "RSA",
+            "kid": "ABCDEFGHIJKLMNOPQRSTUVWXYZ012345",
+            "n": "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789
+            ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEF
+            GHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKL
+            MNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCDEFGHIJKLMNOPQR
+            STUVWXYZ0123456789ABCDEFGHIJKLMNOPQR",
+            "e": "AQAB"
+        }
+    ]
+}
+```
+
+2. Save the JWT portion of the previous command's output to an environmental variable, using the syntax appropriate for the shell you're using:
+```console
+export JWT=0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN012
+3456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOK
+EN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JW
+T-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456
+789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN01
+23456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TO
+KEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789J
+WT-TOKEN0123456789JWT-TOKEN012
+```
+
+This environmental variable will be used by the AWS `aws sts assume-role-with-web-identity` command later.
+
+3. Save the JWT to a file for use by the AWS `aws s3 cp` command later.
+
+```console
+echo JWT=0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN01234
+56789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN
+0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-
+TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN012345678
+9JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123
+456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKE
+N0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT-TOKEN0123456789JWT
+-TOKEN0123456789JWT-TOKEN012 > token
+```
+
+4. Optionally, test that you can retrieve an STS token by running the following command:
+
+```console
+aws sts assume-role-with-web-identity --role-arn arn:aws:iam::012345678901:role/oidc-role \
```

Comment:
Is the value of `--role-arn` always `arn:aws:iam::012345678901:role/oidc-role`? Or do I need to have remembered this from when I created the IAM role previously?